lib/Resolver.js

```diff
@@ -13,11 +13,52 @@ const {
 	PathType
 } = require("./pathUtils");
 
+/** @typedef {import("tapable").Hook} Hook */
+/**
+ * @typedef {Object} ResolveRequest
```

Comment:
It is not all possible request properties. I actually don't know is it required to define one big type since it could be some weird private fields see https://github.com/webpack/enhanced-resolve/blob/master/lib/getInnerRequest.js